ouroboros-consensus/test-storage/Test/Ouroboros/Storage/FS/StateMachine.hs

```diff
@@ -489,6 +493,13 @@ generator Model{..} = oneof $ concat [
          , choose (-1, -10)
          ]
 
+    genOffsetPos :: Gen Int64
+    genOffsetPos = frequency
+         [ (5, return 0)
+         , (3, choose (1, 3))
+         , (2, choose (4, 10))
+         ]
```

Comment:
Why these values? Why not `NonNegative (Small Word64)`, i.e.`genOffset = getNonNegative . getSmall <$> arbitrary`?

Alternatively, the generator could take the file size into account and generate 0, some valid offset, and offsets >= file size.